component/src/action_handler/actions/undelegate.rs

```diff
@@ -36,23 +36,6 @@ impl ActionHandler for Undelegate {
             ));
         }
 
-        // Check whether the end epoch is correct for the given validator (it
-        // may already be unbonding).  This will only differ from the generic
-        // unbonding time if the validator is already unbonding, so we only
-        // require that the declared unbonding period is at least as long as the
-        // expected one, to allow clients the option of skipping a query to
-        // discover if the validator is already unbonding.
```

Comment:
i'm not as keyed into the unbonding token mechanisms, but i think this is the correct change? this PR just removed this validation, and replaced it with a check in `undelegate_claim` that verifies that at least `unbonding_epochs` have passed since the undelegate action was triggered

Reply:
LGTM!